.clang-tidy

```diff
@@ -1,10 +1,24 @@
-Checks: >
-  bugprone-*,
-  performance-*,
-  modernize-*,
-  readability-*,
-  cppcoreguidelines-*
-  -modernize-use-trailing-return-type
+Checks: "*,
```

Comment:
Enabling everything and then opting out of a bunch of categories seems very weird to me. This implies that if new categories of checks get added to clang-tidy, when we upgrade, we'll opt into them without any changes to this file. Am I missing something?

Reply:
@patrickroberts apparently json uses the same style

https://github.com/nlohmann/json/blob/develop/.clang-tidy

Reply:
@patrickroberts that's exactly what it means.  And IMO it's a feature, not a bug.

If we opt-in to checks, then we will miss new ones that are added (until we do considerable legwork).  Provided we have a fixed version of clang-tidy that we're using in CI/CD so upgrades aren't "magical", then we're stable and when we DO upgrade we immediately gain the new checks that we haven't yet violated, and we are told immediately which new checks we're violating.  We can then add the latter as disabled and we've gained new checks for essentially free, and have a clear TODO list that we can prioritize.  win-win.

Reply:
@blozano-tt with the current set, are we able to scan the entire repo and come up clean?

Reply:
@afuller-TT I haven't tried, but I doubt it.